core/src/Kokkos_Serial.hpp

```diff
@@ -212,6 +212,13 @@ class Serial {
 #else
   Kokkos::Impl::HostSharedPtr<Impl::SerialInternal> m_space_instance;
 #endif
+  friend bool operator==(Serial const& lhs, Serial const& rhs) {
+    return lhs.impl_internal_space_instance() ==
+           rhs.impl_internal_space_instance();
+  }
+  friend bool operator!=(Serial const& lhs, Serial const& rhs) {
+    return !(lhs == rhs);
+  }
```

Comment:
What the heck are distinct `Serial` execution space instances?

Reply:
I could imagine a refactor where we actually associate a single thread with the Serial execution space and have a "inline" or "CallingThread" execution space.